### PR TITLE
Add `get_learned_distributions` for `GaussianCopulaSynthesizer` and `CopulaGANSynthesizer`

### LINKED ISSUE #1075: Add get_learned_distributions method to synthesizers
### Problem Description

_As a user, it would be helpful if I could view the distributions and parameters learned by the model for each column and table._

### Acceptance criteria

- There should be a `get_learned_distributions` method added to the `GaussianCopulaSynthesizer` and `CopulaGANSynthesizer`
- The method should return a dictionary mapping the name of the column to the distribution name and the learned parameters for that distribution.
- This method should also be added to the `HMASynthesizer`, but return the dictionaries nested by table name.

### Expected behavior

```python
gc_synthesizer = GaussianCopulaSynthesizer(metadata)
gc_synthesizer.fit(data)
gc_synthesizer.get_learned_distributions()
{
  'age': {
    'distribution': 'beta',
    'learned_parameters': { 'a': 4, 'b': 5 } 
  },
  'weight': {
    'distribution': 'uniform',
    'learned_parameters': { 'loc': 100, 'scale': 200 }
  },
  ...
}
```

Error case: The synthesizer hasn't been fit yet.

```python
gc_synthesizer.get_learned_distributions()
Error: Distributions have not been learned yet. Please fit your synthesizer first using 'fit'.
```

Multi-table case
```python
hma_synthesizer = HMASynthesizer(metadata)
hma_synthesizer.fit(data)

hma_synthesizer.get_learned_distributions('users')
{
  'age': {
    'distribution': 'beta',
    'learned_parameters': { 'a': 4, 'b': 5 }
  },
  'weight': { 
    'distribution': 'uniform',
    'learned_parameters': { 'loc': 10, 'scale': 20 }
  }
}
```
### Additional context

- We should include any intermediary columns we created (for example, `is_null`)

### PR DESCRIPTION
**Pull request by pvk-developer**
Pending tests and RDT version update.
Resolves #1075

### CODE REVIEW

File: tests/unit/single_table/test_copulas.py [hunk 1]
```diff
@@ -389,3 +390,66 @@ def test___set_parameters(self, mock_unflatten_dict, mock_copulas):
...
+                'distribution': 'beta',
+                'learned_parameters': {
+                    'a': 1.0,
+                    'b': 1.0,
+                    'loc': 0.0,
+                    'scale': 0.0
+                }
+            },
+            'one': {
+                'distribution': 'beta',
```
Comment (amontanez24) [inline tests/unit/single_table/test_copulas.py:425]: can we test multiple distribution types?

**Review by amontanez24 — CHANGES_REQUESTED** [review]
The issue also asks for a multi-table version of this method which is essentially just a wrapper around the single-table version. Let's add that and then this should be good!

File: sdv/single_table/copulas.py [hunk 2]
```diff
@@ -178,13 +178,21 @@ def get_learned_distributions(self):
...
         learned_distributions = {}
+        metadata_columns = list(self.metadata._columns)
         for column, learned_params in zip(columns, univariates):
-            distribution = self.numerical_distributions.get(column, self.default_distribution)
-            learned_params.pop('type')
-            learned_distributions[column] = {
-                'distribution': distribution,
-                'learned_parameters': learned_params
-            }
+            for valid_column in metadata_columns:
```
Comment (amontanez24) [inline sdv/single_table/copulas.py:183]: As opposed to looping through these each time, can we do one pass and just get all the valid columns and the just loop through those to set the distributions?

**Review by amontanez24 — COMMENTED** [review]
One comment, then looks good to me!

**Review by fealho — APPROVED** [review]
Besides Andrews comment, looks good👍

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Pull request merged**